common/djangoapps/course_modes/admin.py

```diff
@@ -43,38 +96,111 @@ def clean_course_id(self):
 
         return course_key
 
-    def __init__(self, *args, **kwargs):
-        super(CourseModeForm, self).__init__(*args, **kwargs)
-        if self.instance.expiration_datetime:
-            default_tz = timezone(settings.TIME_ZONE)
-            # django admin is using default timezone. To avoid time conversion from db to form
-            # convert the UTC object to naive and then localize with default timezone.
-            expiration_datetime = self.instance.expiration_datetime.replace(tzinfo=None)
-            self.initial['expiration_datetime'] = default_tz.localize(expiration_datetime)
-
     def clean_expiration_datetime(self):
-        """changing the tzinfo for a given datetime object"""
+        """
+        Ensure that the expiration datetime we save uses the UTC timezone.
+        """
         # django admin saving the date with default timezone to avoid time conversion from form to db
         # changes its tzinfo to UTC
         if self.cleaned_data.get("expiration_datetime"):
             return self.cleaned_data.get("expiration_datetime").replace(tzinfo=UTC)
 
+    def clean_verification_deadline(self):
+        """
+        Ensure that the verification deadline we save uses the UTC timezone.
+        """
+        if self.cleaned_data.get("verification_deadline"):
+            return self.cleaned_data.get("verification_deadline").replace(tzinfo=UTC)
+
+    def clean(self):
+        """
+        Clean the form fields.
+        This is the place to perform checks that involve multiple form fields.
+        """
+        cleaned_data = super(CourseModeForm, self).clean()
+        mode_slug = cleaned_data["mode_slug"]
```

Comment:
Use `cleaned_data.get(...)` here; otherwise, errors will be thrown if, for example, a user only provides a date without a time, or some other invalid input.